Add Fortuno unit testing framework

aradi opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| Rust            | `@conda-forge/help-rust`      |
| Go              | `@conda-forge/help-go`        |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Gitter channel][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://gitter.im/conda-forge/conda-forge.github.io

All apologies in advance if your recipe PR does not receive prompt attention.
This is a high volume repository and the reviewers are volunteers. Review times
vary depending on the number of reviewers on a given language team and may be days
or weeks. We are always looking for more staged-recipe reviewers. If you are
interested in volunteering, please contact a member of @conda-forge/core.
We'd love to have the help!
-->

Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] ~If static libraries are linked in, the license of the static library is packaged.~
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/fortuno/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipes/fortuno/meta.yaml**:

 * ℹ️ License is not an SPDX identifier (or a custom LicenseRef) nor an SPDX license expression.

Documentation on acceptable licenses can be found [here]( https://conda-forge.org/docs/maintainer/adding_pkgs.html#spdx-identifiers-and-expressions ).

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/fortuno/meta.yaml```) and found it was in an excellent condition.

aradi commented:
@conda-forge/staged-recipes, ready for review!

aradi commented:
@conda-forge/staged-recipes, ready for review!

aradi commented:
@conda-forge/staged-recipes, @conda-forge/help-c-cpp ready for review!

aradi commented:
@conda-forge/staged-recipes, @conda-forge/help-c-cpp ready for review!

aradi commented on recipes/fortuno/meta.yaml:
```diff
@@ -0,0 +1,82 @@
+
+requirements:
+  build:
+    - {{ compiler('fortran') }}
```
@ocefpaf I'd really appreciate it!

ocefpaf commented on recipes/fortuno/meta.yaml:
```diff
@@ -0,0 +1,82 @@
+    solutions.
+  license: BSD-2-Clause-Patent
+  license_family: BSD
+  # It is required to include a license file in the package,
```
Please remove these comments. They don't help the recipe and are well defined in conda-forge's docs.

ocefpaf commented on recipes/fortuno/meta.yaml:
```diff
@@ -0,0 +1,82 @@
+requirements:
+  build:
+    - {{ compiler('fortran') }}
+    # this is necessary for all compiled recipes
```
```suggestion
```

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/fortuno/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipes/fortuno/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-recipe-manager`. The recipe can only be automatically migrated to the new v1 format if it is parseable by conda-recipe-manager.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/12585027840](https://github.com/conda-forge/conda-forge-webservices/actions/runs/12585027840). Examine the logs at this URL for more detail.</sub>

aradi commented on recipes/fortuno/build.sh:
```diff
@@ -0,0 +1,19 @@
+BUILD_DIR="_build"
+FFLAGS="-fno-backtrace" cmake "${cmake_options[@]}" -GNinja -B ${BUILD_DIR}
+cmake --build ${BUILD_DIR}
+cmake --install ${BUILD_DIR}
```
Good idea, thanks, I've added it now.

aradi commented:
@ocefpaf Thanks a lot for reviewing. I've removed the superfluous comment lines from `meta.yaml` and added a line to invoke `ctest` after the build in `build.sh`.
